### PR TITLE
Complete upgrading to api version 2024-02-01

### PR DESCRIPTION
**Pull request by Ptnan7**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafCustomRuleObject.cs [hunk 1]
```diff
@@ -77,12 +78,17 @@ public class NewFrontDoorWafCustomRuleObject : AzureFrontDoorCmdletBase
         /// Enabled State. Possible values include: 'Enabled', 'Disabled'
         /// </summary>
         [Parameter(Mandatory = false, HelpMessage = "Enabled State. Possible values include: 'Enabled', 'Disabled'.")]
-        [PSArgumentCompleter("Enabled", "Disabled")]
```
Comment (t-bzhan) [inline src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafCustomRuleObject.cs:80]: Why this got removed?

File: src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafLogScrubbingRuleObject.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "Gets or sets the variable to be scrubbed from the logs.")]
+        [PSArgumentCompleter("RequestHeaderNames", "RequestCookieNames", "QueryStringArgNames", "RequestBodyPostArgNames", "RequestBodyJsonArgNames")]
+        public string MatchVariable { get; set; }
+
+        /// <summary>
+        /// Gets or sets when matchVariable is a collection, operate on the selector to
+        /// specify which elements in the collection this rule applies to.
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "Gets or sets when matchVariable is a collection, operate on the selector to specify which elements in the collection this rule applies to.")]
```
Comment (t-bzhan) [inline src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafLogScrubbingRuleObject.cs:43]: The help message here seem incorrect.

File: src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafCustomRuleObject.cs [hunk 3]
```diff
@@ -77,12 +78,17 @@ public class NewFrontDoorWafCustomRuleObject : AzureFrontDoorCmdletBase
...
         public string EnabledState { get; set; }
 
+        /// <summary>
+        /// Gets or sets describes the list of variables to group the rate limit
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Gets or sets describes the list of variables to group the rate limit")]
```
Comment (t-bzhan) [inline src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafCustomRuleObject.cs:86]: As far as I can see, all the help messages for the new cmdlet and parameters need to be refined. They are publicly available to our customers, so we should make them more readable and understandable. The existing ones do not seem to fulfill this purpose. You could refer to the swagger for some explanations.

File: src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafPolicy.cs [hunk 4]
```diff
@@ -112,6 +114,17 @@ public class NewFrontDoorWafPolicy : AzureFrontDoorCmdletBase
...
 
+        /// Defines rules that scrub sensitive fields in the Web Application Firewall
+        [Parameter(Mandatory = false, HelpMessage = "Defines rules that scrub sensitive fields in the Web Application Firewall.")]
+        public PSFrontDoorWafLogScrubbingSetting LogScrubbingSetting { get; set; }
+
+        /// Defines the JavaScript challenge cookie validity lifetime in minutes. This
+        /// setting is only applicable to Premium_AzureFrontDoor. Value must be an
+        /// integer between 5 and 1440 with the default value being 30.
+        [Parameter(Mandatory = false, HelpMessage = "setting is only applicable to Premium_AzureFrontDoor. Value must be an integer between 5 and 1440 with the default value being 30.")]
+        [ValidateRange(5,1440)]
```
Comment (t-bzhan) [inline src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafPolicy.cs:125]: Let us remove this validation for now, as it is only mandatory for some SKUs.

File: src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafPolicy.cs [hunk 5]
```diff
@@ -142,10 +168,23 @@ public override void ExecuteCmdlet()
...
                     CustomBlockResponseStatusCode = this.IsParameterBound(c => c.CustomBlockResponseStatusCode) ? CustomBlockResponseStatusCode : (int?)null,
                     RedirectUrl = RedirectUrl,
-                    RequestBodyCheck = this.IsParameterBound(c => c.RequestBodyCheck) ? RequestBodyCheck : PSEnabledState.Enabled.ToString()
+                    RequestBodyCheck = this.IsParameterBound(c => c.RequestBodyCheck) ? RequestBodyCheck : PSEnabledState.Enabled.ToString(),
                 },
-                Sku = this.IsParameterBound(c => c.Sku) ? new Management.FrontDoor.Models.Sku(Sku) : null
+                Sku = this.IsParameterBound(c => c.Sku) ? new Management.FrontDoor.Models.Sku(Sku) : null,
             };
+
+            if (JavascriptChallengeExpirationInMinutes >=5 && JavascriptChallengeExpirationInMinutes <= 1440)
```
Comment (t-bzhan) [inline src/FrontDoor/FrontDoor/Cmdlets/NewFrontDoorWafPolicy.cs:176]: As I said in other comment, this is not mandatory for some SKUs, let us remove the validation here and let the service to conduct the validation.

File: src/FrontDoor/FrontDoor/Cmdlets/UpdateFrontDoorWafPolicy.cs [hunk 6]
```diff
@@ -196,7 +208,34 @@ public override void ExecuteCmdlet()
...
+                updateParameters.PolicySettings.JavascriptChallengeExpirationInMinutes = JavascriptChallengeExpirationInMinutes;
+            }
+
+            if (this.IsParameterBound(c => c.LogScrubbingSetting))
+            {
+                var scrubbingRule = new List<Management.FrontDoor.Models.WebApplicationFirewallScrubbingRules>();
+
+                if (LogScrubbingSetting.ScrubbingRule.Count() > 0)
+                {
+                    foreach (var item in LogScrubbingSetting.ScrubbingRule)
```
Comment (t-bzhan) [inline src/FrontDoor/FrontDoor/Cmdlets/UpdateFrontDoorWafPolicy.cs:222]: Could the LogScrubbingSetting be null?

**Review by t-bzhan — CHANGES_REQUESTED** [review]

File: src/FrontDoor/FrontDoor/Models/PSFrontDoorWafCustomRuleGroupByVariable.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.FrontDoor.Models
+{
+    public class PSFrontDoorWafCustomRuleGroupByVariable
+    {
+        public string VariableName { get; set; }
+    }
+
+    public enum PSAzFrontDoorWafCustomRuleGroupByVariableum
```
Comment (t-bzhan) [inline src/FrontDoor/FrontDoor/Models/PSFrontDoorWafCustomRuleGroupByVariable.cs:23]: Do you mean PSAzFrontDoorWafCustomRuleGroupByVariableEnum?

**Comment by NoriZC** [comment]
Please update the [changelog ](https://github.com/Ptnan7/azure-powershell/blob/main/src/FrontDoor/FrontDoor/ChangeLog.md) before merging

File: src/FrontDoor/FrontDoor.Sdk/FrontDoor.Sdk.sln [hunk 8]
```diff
@@ -0,0 +1,25 @@
+﻿
```
Comment (NoriZC) [inline src/FrontDoor/FrontDoor.Sdk/FrontDoor.Sdk.sln:1]: Please remove this .sln file

**Comment by Ptnan7** [comment]
Anything I should do? @NoriZC

**Comment by NoriZC** [comment]
> Anything I should do? @NoriZC

Please update changelog as I mentioned

**Review by NoriZC — APPROVED** [review]

**Pull request merged**